package/MDAnalysis/analysis/hole2/helper.py

```diff
@@ -0,0 +1,556 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2020 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+
+"""HOLE Analysis --- :mod:`MDAnalysis.analysis.hole2.helper`
+=====================================================================================
+
+:Author: Lily Wang
+:Year: 2020
+:Copyright: GNU Public License v3
+
+.. versionadded:: 1.0
+
+Helper functions used in :mod:`MDAnalysis.analysis.hole2.hole`
+"""
+
+from __future__ import absolute_import
+
+import logging
+import tempfile
+import subprocess
+import os
+import numpy as np
+import six
+import errno
+
+from MDAnalysis.lib import util
+from MDAnalysis.exceptions import ApplicationError
+from .templates import (SIMPLE2_RAD, IGNORE_RESIDUES, hole_input,
+                        hole_lines, exe_err)
+
+logger = logging.getLogger(__name__)
+
+
+def write_simplerad2(filename="simple2.rad"):
+    """Write the built-in radii in :data:`SIMPLE2_RAD` to `filename`.
+
+    Does nothing if `filename` already exists.
+
+    Parameters
+    ----------
+    filename : str, optional
+       output file name; the default is "simple2.rad"
+
+    Returns
+    -------
+    filename : str
+       returns the name of the data file
+    """
+
+    if not os.path.exists(filename):
+        with open(filename, "w") as rad:
+            rad.write(SIMPLE2_RAD + "\n")
+        logger.debug("Created simple radii file {}".format(filename))
+    return filename
+
+
+def check_and_fix_long_filename(filename, tmpdir=os.path.curdir,
```

Comment:
This function is not well tested, see coverage. (It wasn't well tested before, either.)

Reply:
Added a test class.